src/couch/src/couch_httpd_multipart.erl

```diff
@@ -134,6 +138,10 @@ mp_abort_parse_atts(_, _) ->
 
 maybe_send_data({Ref, Chunks, Offset, Counters, Waiting}) ->
     receive
+        {hello_from_writer, Ref, WriterPid} ->
+            WriterRef = erlang:monitor(process, WriterPid),
+            NewCounters = orddict:store(WriterPid, {WriterRef, 0}, Counters),
```

Comment:
@nickva now that you pointed out the existence of multiple writers for a given `WriterPid` I think this logic to initialize the count to 0 for each writer is wrong. I I think it will cause `maybe_send_data` to use the wrong index when sending chunks to writers of attachments 2..N, because the index is on the set of all chunks for the entire HTTP request. You agree?

Reply:
Yeah, agree. It seems we'd want to initialize each separate follows parser only once. Then I think it could work? 

So, for example, we could fold over all docs [1] and all the attachments and keep a map of `#{ParserPid => Ref}`. If the `ParserPid` is not in the map, we send the `{hello_from_writer, Ref, WriterPid}` and start monitoring it. Then we use its `Ref` in the `receive` `DOWN'` inside the closure. If, `ParserPid` is in the map ,then we skip sending the `hello` message and just use its existing `Ref` in the `receive DOWN'`.

[1] It's seems it's not just a doc with multiple attachment but it could also be multiple docs, each with multiple attachments?